### ISSUE TITLE
Simplified SSL cert configuration

### ISSUE THREAD

**Issue by OnePressTech** [turnkeylinux/tracker#546]
[this issues is (somewhat) covered by #382]

It would be nice if the configuration console supported a SSL certificate installation and that this installation extended to the O/S and the utilities like Webmin. I made the mistake of attempting to adjust the cert.pem as per an old TKLX blog post suggestion and ended up in systemd no-man's land for a while (can't find d-bus...ahhh....bloody systemd!!) until I figured out how to back out the change in the almost-bricked VM instance.

The issue (feature enhancement) is for simpler support for commercial SSL certs for both the site and the tools including VM initialisation time and yearly renewal time. A suggested initial focus would be on the [Let's Encrypt](https://letsencrypt.org/) initiative that offers a free SSL certificate that does not pop up with the Browser warning that a self-signed certificate gets.
##

**Comment by JedMeister** [comment]
I agree. For v14.1 we're hoping to implement #382 which sort of ties in with this; but not quite...

**Comment by Dude4Linux** [comment]
We do need to do a better job of documenting and updating documentation.  Relying on old blog posts is a recipe for disaster.

**Comment by OnePressTech** [comment]
Thanks John...couldn't agree more :-)

Jed...any possibility you and the TKLX boys could choose a collaborative document model (Googledocs shared doc, slack area, wiki, github) for us to work together on a install guide, admin guide, user guide combo? I can commit time to documentation immediately and on an ongoing basis...though I would not be expecting to write them alone.

**Comment by Dude4Linux** [comment]
@OnePressTech  FWIW, v14.0 includes a new utility turnkey-make-ssl-cert.

    turnkey-make-ssl-cert ver. 1.2
    "Make server cert for TurnKey GNU/Linux appliance"
    
    Usage: turnkey-make-ssl-cert [-o|--out file] [-t|--template file] [-i|--ip] [-v|--verbose] [-f|--force-overwrite] FQDN .. [FQDN]
      Generate a certificate/key pair using the list of FQDNs.
    
    Usage: turnkey-make-ssl-cert [-d|--default] [-t|--template file] [-i|--ip] [-v|--verbose] [-f|--force-overwrite]
      Generate the default certificate/key pair, cert.crt, cert.key, using the hostname.
    
    Usage: turnkey-make-ssl-cert [-o|--out file] [-t|--template file] [-i|--ip] [-v|--verbose] [-f|--force-overwrite] [-w|--wild] domainName .. [domainName]
      Generate a wildcard certificate for the list of domains.
    
    Usage: turnkey-make-ssl-cert [-d|--default] [-t|--template file] [-i|--ip] [-v|--verbose] [-f|--force-overwrite] [-r|--csr] FQDN .. [FQDN]
      Generate an optional certificate signing request for the list of FQDNs.
    
    Usage: turnkey-make-ssl-cert [-h|--help]
      Display the help message and exit.
    
          Options:
          -h, --help              Display this help message and exit
          -o, --out [/path/]file  Write certificate to alternate location
          -d, --default           Generate default certificate
                                    /etc/ssl/private/cert.pem
          -r, --csr               Generate a certificate signing request
          -w, --wild              Generate wildcard certificate
          -t, --template file     Use alternate template file
                                    default: /etc/ssl/turnkey.cnf
          -i, --ip                Optionally include host ip addresses
          -v, --verbose           Display generated certificate
          -f, --force-overwrite   Overwrite existing certificate
    
          NOTE: You must be the superuser to run this script.

To create a certificate request, you can either edit the turnkey.cnf or create a custom file and use the '--template' option along with '--csr'.  It will then generate a temporary self-signed certificate key pair along with the csr request.

**Comment by OnePressTech** [comment]
Thanks John.

The issue (feature enhancement is for simpler support for commercial SSL certs for both the site and the tools including VM initialisation time and yearly renewal time. A suggested initial focus would be on the [Let's Encrypt](https://letsencrypt.org/) initiative that offers a free SSL certificate that does not pop up with the Browser warning that a self-signed certificate gets.

Apologies for the confusion. I will edit the original issue to make it clearer.

**Comment by Dude4Linux** [comment]
@OnePressTech I haven't had a chance to talk to Jeremy and the other principles yet but I think using Read The Docs or the software it's based on, Sphinx, is the way to go. Just yesterday I was reading an article recommending Sphinx for documentation over a Wiki. ReadTheDocs is a cloud solution that has some advantages (better search engine) than a pure Sphinx application.  The decision to move in that direction would have to be made by Liraz, Alon, and Jeremy.

https://read-the-docs.readthedocs.org/en/latest/getting_started.html

**Comment by OnePressTech** [comment]
Thanks John..that looks interesting..much appreciated :-)

**Comment by JedMeister** [comment]
>>We do need to do a better job of documenting and updating documentation. Relying on old blog posts is a recipe for disaster.

>Thanks John...couldn't agree more :-)

I agree as well. Documentation has always been the weakest part of TurnKey IMO. Over the years (at times) I have made significant effort to improve that situation, however it has never been that great.

>Jed...any possibility you and the TKLX boys could choose a collaborative document model (Googledocs shared doc, slack area, wiki, github) for us to work together on a install guide, admin guide, user guide combo? I can commit time to documentation immediately and on an ongoing basis...though I would not be expecting to write them alone.

The website docs section (https://www.turnkeylinux.org/docs) is a wiki. So that may be a good place to start for the moment. However I agree with John that Read The Docs is a great potential option.

>NOTE: We should probably create another issue around collaborative docos though to keep this issue focused on an auto-ssl config.

Agreed and done: #555

**Comment by JedMeister** [comment]
As per https://github.com/turnkeylinux/tracker/issues/369#issuecomment-261474743 I am assigning @DocCyblade as lead battle tester! ￼:smile:

**Comment by DocCyblade** [comment]
I'll post my testing methodology and testing list here in this post. For now I am adding some info for my self, as I am new to Let's Encrypt looks nice. This will be tested with the new confconsole that needs to be tested as a plug-in that was written.




For my Info:

How LE works:
https://letsencrypt.org/how-it-works/

LE Stage/Testing:
https://letsencrypt.org/docs/staging-environment/

**Comment by DocCyblade** [comment]
@qrntz - Was having a go with the Lets Encrypt plug in, and not sure if it's a bug or how I am running it.

So on a fresh install of core, stopped the service,  I removed the `/usr/lib/confconsole` directory and cloned the confconsole repo right to it `/usr/lib/confconsole` then started the service. 

I can see the new menu items :-) 

I tried the Lets Encrypt -> Letsencrypt menu item, and it won't let me Cancel. I hit escape, it looks like it quits the confconsole.
